File: components/OS.java
package components; // ← Fará parte do pacote "components", junto com outras peças de computador.
// ↓ Criando a classe OS (sistema operacional)
public class OS {
 private String nome;
 private Integer bit; // ← 32 ou 64 bits?
 private Integer gb; // ← Ocupação no armazenamento
 // ↓ Argumentos para criar a classe
 public OS (String nome, Integer bit, Integer gb) {
  // ↓ NOTA: Os tratadores de exceção abaixo, marcados com if, são personalizados.
  if (nome == null || nome.trim().isEmpty()) {throw new IllegalArgumentException("Não pode ser vazio ou nulo!");}
  if (bit != 32 && bit != 64) {throw new IllegalArgumentException("Deve ser 32-bit ou 64-bit!");}
  if (gb <= 0) {throw new IllegalArgumentException("Deve ser maior que zero!");}
  this.nome=nome;
  this.bit=bit;
  this.gb=gb;
 }
 // ↓ Simplesmente digitando o nome do instance dentro do parênteses no "System.out.println()" irá renderizar a string abaixo
 @Override
 public String toString() {
  return "\n::::: PROPRIEDADES DO SISTEMA OPERACIONAL :::::\nNome: " +nome+ "\nBits: " +bit+ "-bit\nOcupação: " +gb+ " GB\n";
 }
 // ↓ NOTA: Para manter os atributos da classe como private e para serem visíveis por outras classes, os getters abaixo serão utilizados.
 public String verNome() {return nome;}
 public Integer verBits() {return bit;}
 public Integer verOcupacao() {return gb;}
}

File: components/OS.py
# ↓ Criando a classe OS (sistema operacional)
class OS:
 bits=[32,64]
 # ↓ Argumentos para criar a classe
 def __init__(self, nome:str, bit:int, gb:int):
  
  # ↓ NOTA: Os tratadores de exceção abaixo, marcados com if, são personalizados.
  if nome is None or nome.strip() == "":
   raise ValueError("Não pode ser vazio ou nulo!")
  if bit not in OS.bits:
   raise ValueError("Deve ser 32-bit ou 64-bit!")
  if gb<=0:
   raise ValueError("Deve ser maior que zero!")
  
  self.__nome=nome
  self.__bit=bit # ← 32 ou 64 bits?
  self.__gb=gb # ← Ocupação no armazenamento
  
  # ↓ Simplesmente digitando o nome do instance no console, ou dentro do parênteses no "print()" irá renderizar a string abaixo
 def __repr__(self) -> str:
  return f"\n::::: PROPRIEDADES DO SISTEMA OPERACIONAL :::::\nNome: {self.__nome}\nBits: {self.__bit}-bit\nOcupação: {self.__gb} GB\n"
  
  # ↓ NOTA: Para manter os atributos da classe como privados e serem visíveis por outras classes, os getters abaixo serão utilizados.
 def verNome(self) -> str:
  return self.__nome
 def verBits(self) -> int:
  return self.__bit
 def verOcupacao(self) -> int:
  return self.__gb
